restapi/src/main/java/io/stargate/web/docsapi/service/DocumentService.java

```diff
@@ -1219,6 +1231,25 @@ private String getParentPathFromRow(Row row) {
     return s.toString();
   }
 
+  private String getFieldPathFromRow(Row row) {
+    boolean pathComplete = false;
+    int i = 0;
+    StringBuilder fullPath = new StringBuilder();
+    while (!pathComplete) {
+      String value = row.getString("p" + i);
+      if (value.isEmpty()) {
+        pathComplete = true;
+      } else {
+        if (i != 0) {
+          fullPath.append(".");
+        }
+        fullPath.append(value);
+      }
+      i++;
+    }
+    return fullPath.toString();
```

Comment:
`Nit:` you may use Guava's Joiner.

```suggestion
     List<String> path = new ArrayList<>();
     for (int i = 0; i < DocumentDB.MAX_DEPTH; i++) {
           String value = row.getString("p" + i);
           if (value.isEmpty()) {
              break;
           }
           path.add(value);
     }
     return Joiner.on(".").join(path);
```
```